monai/transforms/lazy/functional.py

```diff
@@ -26,17 +26,83 @@
 )
 from monai.utils import LazyAttr, look_up_option
 
-__all__ = ["apply_transforms"]
+__all__ = ["apply_pending", "apply_pending_transforms"]
 
 __override_keywords = {"mode", "padding_mode", "dtype", "align_corners", "resample_mode", "device"}
 
 
-def apply_transforms(
-    data: torch.Tensor | MetaTensor, pending: list | None = None, overrides: dict | None = None, **kwargs: Any
+def apply_pending_transforms(data, overrides: dict | None = None, logger_name: str | None = None):
+    """
+    apply_pending_transforms iterates over a tuple, list, or dictionary of data, recursively calling itself
+    to get a single tensor. If that tensor is a MetaTensor with pending lazy transforms, it then calls
+    ``apply_pending_to_tensor`` on each element to perform the executing of the pending transforms.
+
+    This method optionally takes a set of overrides that can be used to change specific parameters on the
+    transform pipeline. See ``Compose`` for more details. This method takes a logger_name that can be used
+    to override the default logger, to provide telemetry during the execution of pending transforms.
+
+    This method is intended primarily for use by ``execute_compose`` and other methods that handle the
+    underlying execution of transform pipelines. You should not need to use it in the general case, unless
+    you are developing functionality to perform such operations.
+
+    Args:
+        data: a ``torch.Tensor`` or ``MetaTensor``, or list, tuple or dictionary of tensors.
+        overrides: An optional dictionary that specifies parameters that can be used to override transform
+            arguments when they are called
+        logger_name: An optional name for a logger to be used when applying pending transforms. If None,
+            logging is suppressed.
+    Returns:
+
+    """
+    if isinstance(data, list):
+        return [apply_pending_transforms(d, logger_name=logger_name) for d in data]
+
+    if isinstance(data, tuple):
+        return tuple(apply_pending_transforms(d, logger_name=logger_name) for d in data)
+
+    if isinstance(data, dict):
+        d = data
```

Comment:
Is it needed to make a fresh dict here and not modify the input?
```suggestion
        d = dict(data)
```

Reply:
Ahh, this was actually a fix from
```
d = dict(data)
```
specifically to 
```
d = data
```
because tests were expecting id compatibility. it is true that `data` could just be used throughout instead

Reply:
Should the tests be changed then or is this a property that we want? It needs to be commented here and (if not already) documented that this mutates the input dictionary.

Reply:
Ahh, I just need to move the dict copy inside the conditional